plugins/json_processor/src/json_processor/component.py

```
@@ -0,0 +1,228 @@
+                sample_size = min(self.config.sample_size, len(data))
+                result = data.sample(n=sample_size)
+                logger.debug(f"Applied sampling: {sample_size} rows")
+                return result
+            except Exception as e:
+                logger.warning(f"Failed to apply size sampling: {e}")
+
+        elif self.config.sample_fraction:
+            try:
+                result = data.sample(fraction=self.config.sample_fraction)
+                logger.debug(
+                    f"Applied fraction sampling: {self.config.sample_fraction}"
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)